src/utils/fetch.ts

```diff
@@ -28,15 +28,14 @@ export const withTimeout = async (
   return res
 }
 
-// TODO: put in env vars for staging
-export const fetchUntilSuccess = async (endpoints: string[]): Promise<any> => {
-  try {
-    // Pick a random endpoint from the allowed endpoints
-    const endpoint = endpoints[Math.floor(Math.random() * endpoints.length)]
-    console.info('Attempting endpoint: ', endpoint)
-    return await fetchWithTimeout(endpoint)
-  } catch (e) {
-    console.error(e)
-    return await fetchUntilSuccess(endpoints)
-  }
+export const fetchWithLibs = async (req: {
+  endpoint: string
+  queryParams?: object
+}) => {
+  const data = await window.audiusLibs.discoveryProvider._makeRequest(req)
```

Comment:
I think we should avoid window.audiusLibs here.

We can probably also follow the similar pattern in AudiusClient which has libs already as a reference